### PR TITLE
feat(autoscaling): add support for InstanceRefresh suspended process

### PR DESCRIPTION
**Pull request by mulmar-aws**
Also add InstanceRefresh as a default suspended process for RollingUpdate. 

I have also submitted a request to update https://repost.aws/knowledge-center/auto-scaling-group-rolling-updates to conform with this change. 

### Reason for this change

Instance Refresh is a feature of ASG. It performs a similar function to Rolling Update. If an Instance Refresh is running at the same time as a Rolling Update, the Rolling Update will fail. It is safer to suspend the process. 

### Description of changes

See above.

### Description of how you validated changes

Unit tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-autoscaling/test/scheduled-action.test.ts [hunk 1]
```diff
@@ -104,6 +104,7 @@ describeDeprecated('scheduled action', () => {
                 'AZRebalance',
                 'AlarmNotification',
                 'ScheduledActions',
+                'InstanceRefresh'
```
Comment (xazhao) [inline packages/aws-cdk-lib/aws-autoscaling/test/scheduled-action.test.ts:107]:
```suggestion
                'InstanceRefresh',
```

Comment (mulmar-aws) [reply packages/aws-cdk-lib/aws-autoscaling/test/scheduled-action.test.ts:107]: 🤦  - Thanks, I'll fix.

**Comment by xazhao** [comment]
Hello @mulmar-aws there are other tests failing which is because of missing `InstanceRefresh`. Could you please add it to the failing test? Also is integration test needed for this PR?

**Comment by mulmar-aws** [comment]
I think I have the integration test set up properly but I see some automated checks are failing. 

```
> yarn integ --update-on-failed aws-autoscaling/test/integ.asg-update-policy.js
yarn run v1.22.21
$ integ-runner --language javascript --update-on-failed aws-autoscaling/test/integ.asg-update-policy.js

Verifying integration test snapshots...

  UNCHANGED  aws-autoscaling/test/integ.asg-update-policy 0.538s

Snapshot Results:

Tests:    1 passed, 1 total

Running integration tests for failed tests...

Running in parallel across regions: us-east-1, us-east-2, us-west-2

Test Results:

Tests:    0 passed, 0 total
✨  Done in 1.25s.
```

**Comment by mulmar-aws** [comment]
Hello, just following up to see if more changes are required?

**Comment by xazhao** [comment]
The change looks good to me. Just one question: should the title be `feat` instead of `fix`?

**Comment by mulmar-aws** [comment]
Good point. Changed.

**Comment by mulmar-aws** [comment]
Exemption Request : I don't think there is anything to change in a README file here - suspended processes are not called out in detail in the AutoScaling README.

**Review by xazhao — APPROVED** [review]
Looks good to me

**Review by xazhao — APPROVED** [review]

**Pull request merged**